storage/remote/queue_manager.go

```diff
@@ -367,6 +385,31 @@ func (t *QueueManager) Start() {
 	t.metrics.minNumShards.Set(float64(t.cfg.MinShards))
 	t.metrics.desiredNumShards.Set(float64(t.cfg.MinShards))
 
+	// Checking if the JSON file is corrupted
+	var check Checkpoints
+
+	jsonFile, err := os.Open("SegmentRecord.json")
+	defer jsonFile.Close()
+
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	err = json.Unmarshal(byteValue, &check)
+
+	if err != nil {
+		fmt.Println(err)
+	} else {
+		for _, record := range check.Checkpoints {
+			segmentNumber, err := strconv.Atoi(record.Segment)
+			if err != nil {
+				fmt.Println(err)
+			}
+
+			if segmentNumber > -1 && segmentNumber < 100000000 {
```

Comment:
we could define these values somewhere `invalidSegmentNumber` and `maxSegmentNumber`, though I'd check the WAL package files to see if anything similar already exists?